controllers/argocd/configmap.go

```diff
@@ -789,3 +793,41 @@ func (r *ReconcileArgoCD) reconcileGPGKeysConfigMap(cr *argoproj.ArgoCD) error {
 	argoutil.LogResourceCreation(log, cm)
 	return r.Client.Create(context.TODO(), cm)
 }
+
+// reconcileArgoCmdParamsConfigMap will ensure that the ConfigMap containing command line parameters for ArgoCD is present.
+func (r *ReconcileArgoCD) reconcileArgoCmdParamsConfigMap(cr *argoproj.ArgoCD) error {
+	cm := newConfigMapWithName(common.ArgoCDCmdParamsConfigMapName, cr)
+	cm.Data = make(map[string]string)
+
+	if err := controllerutil.SetControllerReference(cr, cm, r.Scheme); err != nil {
+		return err
+	}
+	existingCM := &corev1.ConfigMap{}
+	if argoutil.IsObjectFound(r.Client, cr.Namespace, cm.Name, existingCM) {
+		changed := false
+		if !reflect.DeepEqual(cm.Data, existingCM.Data) {
+			existingCM.Data = cm.Data
+			changed = true
+		}
```

Comment:
We’ll either need to allow the data change directly in the ConfigMap or provide a new field in the ArgoCD CR to update the data in ConfigMap.

Reply:
I will add a new key to the ArgoCD CR to manage the data in the ConfigMap. I believe allowing direct/manual edits to the ConfigMap is not ideal? WDYT @svghadi 
New field in CR -
```
spec:
  cmdParams:
    application.namespaces: "default"
    redis.compression: gzip
```

Reply:
Yes. Makes sense to disallow direct edits. 

What do you think about below structure? It would allow us to move the extraConfig field (which currently corresponds to `argocd-cm`) here in a future release. Right now, we have a number of fields scattered across the ArgoCD CRD, and I am thinking of reorganizing them into a more grouped and structured format.
```yaml
spec:
  configMaps:
    argocdCmdParamsCM:
      application.namespaces: "default"
      controller.log.level: "debug"
    argocdCM:
      timeout.reconciliation: "180s"
      kustomize.buildOptions: "--load-restrictor LoadRestrictionsNone"
```

@anandf @jgwest - Thoughts?

Reply:
Let's go with `spec.cmdParams` for now to unblock things. I will put together a proposal for reorganizing the fields, and we can revisit the discussion then.